PowerPointLabs/PowerPointLabs/ThisAddIn.cs

```diff
@@ -1033,34 +1075,41 @@ private void ThisAddInPresentationClose(PowerPoint.Presentation pres)
                 ShapePresentation.Close();
             }
 
-            // special case: if we are closing ShapeGallery.pptx, no other action will be done
-            if (pres.Name.Contains(ShapeGalleryPptxName))
+            // special case: if we are closing 'ShapeGallery.pptx' or 'Sync Lab - Do not edit.pptx', no other action will be done
+            if (pres.Name.Contains(ShapeGalleryPptxName) || pres.Name.Contains(SyncLabPptxName)) 

```

Comment:
Just wondering would ThisAddInPresentationClose be triggered when picture slides lab's window is closed?

Reply:
Yes, it would be triggered. In this case, we are only checking for ShapesLab and SyncLab's background ppts here because we don't want `ShutDownSyncLab()` and `ShutDownShapesLab()` function to be called more than once. It's alright for PictureSlidesLab because we have a way to track whether PictureSlidesLab is open through Ribbon1, so that logic is already handled in `ShutDownPictureSlidesLab()`.

The behaviour of the background ppts for all three labs differ slightly, hence all these closing methods for each lab also differ slightly. For instance, for PSL the background ppt is closed upon closing of the PSL window. For sync & shapes lab, the background ppt remains open till the end once user clicks on the lab from the ribbon.

Reply:
That sounds good, thanks! :)